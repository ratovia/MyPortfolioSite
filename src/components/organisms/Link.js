import React from "react"
import { CircleImageIcon } from "../atoms/CircleImageIcon"
import { FrederickaH2 } from "../atoms/FrederickaText"
import styled from "styled-components"

const LinkArea = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 60px 30px;
`

const TitleText = styled.div`
  align-self: flex-start;
  margin-bottom: 60px;
`

const LinkListTop = styled.div`
  align-self: flex-start;
  display: flex;
  justify-content: space-around;
`
const LinkListMiddle = styled.div`
  align-self: center;
  display: flex;
  justify-content: space-around;
`
const LinkListBottom = styled.div`
  align-self: flex-end;
  display: flex;
  justify-content: space-around;
`

const Container = styled.div`
  padding: 10px 40px;
`

export const Link = ({ links }) => {
  return (
    <LinkArea>
      <TitleText>
        <FrederickaH2 label="Links" color="white"></FrederickaH2>
      </TitleText>
      <LinkListTop>
        {links.slice(0, 3).map(link => {
          return (
            <Container key={link.name}>
              <CircleImageIcon
                imageUrl={link.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </LinkListTop>
      <LinkListMiddle>
        {links.slice(3, 6).map(link => {
          return (
            <Container key={link.name}>
              <CircleImageIcon
                imageUrl={link.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </LinkListMiddle>
      <LinkListBottom>
        {links.slice(6, 9).map(Link => {
          return (
            <Container key={Link.name}>
              <CircleImageIcon
                imageUrl={Link.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </LinkListBottom>
    </LinkArea>
  )
}
