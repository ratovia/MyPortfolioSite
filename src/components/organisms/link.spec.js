import React from "react"
import renderer from "react-test-renderer"
import backgroundImg from "../../stories/assets/bg.jpg"

import { Link } from "./Link"
describe("Links", () => {
  it("renders correctly", () => {
    const links = [
      {
        name: "Ruby on Rails",
        imageUrl: backgroundImg,
      },
      {
        name: "HTML/CSS",
        imageUrl: backgroundImg,
      },
      {
        name: "JavaScript",
        imageUrl: backgroundImg,
      },
      {
        name: "Python",
        imageUrl: backgroundImg,
      },
      {
        name: "PHP",
        imageUrl: backgroundImg,
      },
      {
        name: "AWS",
        imageUrl: backgroundImg,
      },
      {
        name: "3DCG",
        imageUrl: backgroundImg,
      },
      {
        name: "linux",
        imageUrl: backgroundImg,
      },
      {
        name: "React",
        imageUrl: backgroundImg,
      },
    ]
    const tree = renderer.create(<Link links={links}></Link>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
