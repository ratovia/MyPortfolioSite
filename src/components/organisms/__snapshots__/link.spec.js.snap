// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renders correctly 1`] = `
<div
  className="sc-jRQAMF kLBZMp"
>
  <div
    className="sc-gKckTs jJNaX"
  >
    <h2
      className="sc-hKwCoD ciajrW"
    >
      <p
        className="sc-gsDJrp fFMVcA"
        color="white"
      >
        Links
      </p>
    </h2>
  </div>
  <div
    className="sc-iCfLBT iUFcXm"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
  <div
    className="sc-furvIG dPzpIM"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
  <div
    className="sc-pVTma HwUCx"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
</div>
`;
